### PR TITLE
[Javascript] New exercise: equality

### PR DESCRIPTION
**Pull request by shubhsk88**
The PR addresses issue #2732  which is the implementation of the new exercise for equality  in javascript

### CODE REVIEW

File: languages/javascript/concepts/equality/about.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Javascript uses `==`, `===` and `Object.is()` to compare the equality in different instances
```
Comment (pertrai1) [inline languages/javascript/concepts/equality/about.md:1]: This should have a period at the end. I also have to wonder about the use of instances here. When I read this I am wondering what instances are being referred to. Is this instance for use in the sentence or are you talking about instances from the programming level perspective?

File: languages/javascript/concepts/equality/about.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+In general as conclusion
```
Comment (pertrai1) [inline languages/javascript/concepts/equality/about.md:3]: I don't follow what this means?

File: languages/javascript/concepts/equality/about.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+- If you know the type(s) in a comparison.If both of the types are same then `==` is identical to `===`.So prefering `==` will be shorter to write in such scenario
```
Comment (pertrai1) [inline languages/javascript/concepts/equality/about.md:5]: Would these be better off as a bullet list? Also will need to check spacing between a period and the next word.

File: languages/javascript/concepts/equality/about.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+- If you don't know the type(s) in a comparison. It is preferable to use `===`.
+
+Sometimes implicit coercion can be helpful.In such scenarion the understanding of coercion can be very helpful
```
Comment (pertrai1) [inline languages/javascript/concepts/equality/about.md:8]: check spelling on scenarion

File: languages/javascript/concepts/equality/about.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+You can find all the rules of the equality in [ECMA docs][ecma-docs] and learn more about coercion and equality from [You Don't know JS][ydkjs]
```
Comment (pertrai1) [inline languages/javascript/concepts/equality/about.md:10]: of the equality -> of equality

File: languages/javascript/concepts/equality/introduction.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
+**Equality** in the javascript can be referred in two ways which is `==` or `===`
```
Comment (pertrai1) [inline languages/javascript/concepts/equality/introduction.md:1]: Suggestion is to change this sentence. Recommendation:

**Equality** in JavaScript can be referred in two ways:  `==` or `===`

File: languages/javascript/concepts/equality/introduction.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+It is usually a misconception that people consider `==` which is usually check for value equality and `===` which usually check for the equality and type of the comparison which is not true.
```
Comment (pertrai1) [inline languages/javascript/concepts/equality/introduction.md:2]: This starts off by saying that there is a misconception but does not say what that misconception is? This is also a run on sentence that would be easier to understand if broken down into smaller sentences.

File: languages/javascript/concepts/equality/introduction.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+The usual description as define in the specification of the javascript is that
```
Comment (pertrai1) [inline languages/javascript/concepts/equality/introduction.md:4]: The usual description as define in the specification of the javascript is that

The description defined in the specification of JavaScript is that

File: languages/javascript/concepts/equality/introduction.md [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+`==` allows coercion in the equality while `===` disallow coercion
```
Comment (pertrai1) [inline languages/javascript/concepts/equality/introduction.md:5]: pluralize disallow

File: languages/javascript/concepts/equality/introduction.md [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+
+`==` is usually defined as the abstract equality operator. **This equality prefers Numeric coercion**
```
Comment (pertrai1) [inline languages/javascript/concepts/equality/introduction.md:7]: I would avoid "usually". That can be misleading to a reader as to wonder in which case is it "usually" defined and which case it is not "usually" defined.

File: languages/javascript/concepts/equality/introduction.md [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+
+It follows certain rules while allowing coercion `x==y`:
+
+- If the type of `x` is same as type of `y`. This Equality return the output of `===`
```
Comment (pertrai1) [inline languages/javascript/concepts/equality/introduction.md:11]: This Equality return -> This equality check returns

**Comment by pertrai1** [comment]
If I may, I would suggest that a proofreader take a look at the content in more depth.

**Comment by ErikSchierboom** [comment]
This Friday (Jan 29th 2021), we're going to explode this repo back into the individual track repositories and archive this repo.

As such, this PR needs to be merged by Friday 9am UTC, or it will need to be manually reopened against the track repo.

File: languages/javascript/concepts/equality/about.md (hunk 5 above)
Comment (shubhsk88) [reply languages/javascript/concepts/equality/about.md:10]: Thank you so much for proof reading it

Comment (shubhsk88) [reply languages/javascript/concepts/equality/about.md:10]: I will make changes soon

**Comment by SleeplessByte** [comment]
@shubhsk88 don't worry, it will not be "wasted", if this is not merged before friday.

**Pull request closed without merging**